### DISCUSSION TITLE
Export to Flatpak (Linux-only)

### DISCUSSION THREAD

**Discussion by tseli0s** [godotengine/godot-proposals#5011] [Platforms]
# Export to Flatpak
It has already been mentioned in the Godot repository but it was closed because this repository was opened and it's not a bug. 

Godot plays well with Linux. It's really years ahead of it's Linux competition because there has been focus on it. However, as the time progresses, Linux is slowly transitioning from native packages and raw ELF binaries to Flatpaks and AppImages. The latter one is questionable for a game so I'm only going to cover flatpaks.

Because I don't want to dig into what flatpaks really are, we can just say that they provide a cross-distro way of packaging software. That means I can package one game, and any distribution with flatpak support (More or less all of them have) will be able to run it the same. I personally compare them to APKs since they are similar.

# Reasons to have a Flatpak export option
- Like I said, they work across distributions.
- They provide sandboxing and permission checks.
- Library issues no longer exist (This is technically fixed by Godot already, but you never know)
- A game can be uploaded in a repository like Flathub where any user can download it without much hassle.
- Godot already has the infrastructure required for such an option. Things like XML / JSON editing either already exist or will not be hard to implement with the help of a library. Not much more needed except some presets that will be modified.
# Reasons not to have Flatpak export option
- Maybe the functionality can be added through a plugin instead. If that's the case, the community can develop the option instead. However cross-platform support may be questionable.
- Flatpaks may sometimes be slower or not have support for the latest OpenGL ES versions (Especially for Nvidia graphics cards or old Mesa libraries).
- Simply too much work. Maybe it will take a lot of effort to do it which could be used somewhere else.

**Comment by Calinou** [comment]
I think this is best handled by third-party solutions, as exporting to a Flatpak can only be done from Linux. To my knowledge, `flatpak-builder` doesn't run on Windows and macOS to my knowledge. However, a lot of people are exporting for Linux from Windows and macOS.

Besides, Godot binaries are portable enough to work on pretty much any desktop Linux distribution (and most server distributions, for headless/server binaries).[^1]

[^1]: One exception is Alpine Linux, which requires using a glibc version of the distribution, or a self-compiled Godot build linked against musl. Alpine is mainly used within containers anyway, and isn't used on desktop platforms.

> - Library issues no longer exist (This is technically fixed by Godot already, but you never know)

Indeed, I don't think library issues will suddenly start popping up again. If they did, Flatpak would most likely also be in trouble anyway.

> - Flatpaks may sometimes be slower or not have support for the latest OpenGL ES versions (Especially for Nvidia graphics cards or old Mesa libraries).

Godot doesn't use OpenGL ES on desktop platforms, as the proprietary graphics drivers don't support it directly (including any driver on Windows). Instead, OpenGL is used on desktop platforms via the `GLES_OVER_GL` compile-time define.
